test_omp_target_aligned_alloc_device.c violates "traits-array modifier must be a constant array" restriction

The following code is invalid as it violates a restriction:

https://github.com/OpenMP-Validation-and-Verification/OpenMP_VV/blob/f9cb9f24c4e8909ec5e49894524bead14ca58f69/tests/5.1/allocate/test_omp_target_aligned_alloc_device.c#L24-L27

GCC rejects this with:
```C
tests/5.1/allocate/test_omp_target_aligned_alloc_device.c: In function ‘test_omp_aligned_alloc_on_device’:
tests/5.1/allocate/test_omp_target_aligned_alloc_device.c:27:57: error: traits array ‘traits’ must be of ‘const omp_alloctrait_t []’ type
   27 |   #pragma omp target map(tofrom: errors) uses_allocators(alloc(traits))
      |                                                         ^
```

This violates:
> 2.12.5 target Construct
> …
> Restrictions
> … 
> * Arrays that contain allocator traits that appear in a **uses_allocators** clause must be constant arrays, have constant values and be defined in the same scope as the construct in which the clause appears.

→ OpenMP 5.0, "2.12.5 target Construct", OpenMP 5.1, "2.14.5 target Construct"

In TR14, "13.8 uses_allocators Clause", the wording is only slightly different:
> Restrictions
> …
> * [C / C++] The traits argument for the traits-array modifier must be a constant array, have constant values and be defined in the same scope as the construct on which the clause appears.
> * [Fortran] The traits argument for the traits-array modifier must be a named constant of rank one.

--- a/tests/5.1/allocate/test_omp_target_aligned_alloc_device.c
+++ b/tests/5.1/allocate/test_omp_target_aligned_alloc_device.c
@@ -21,8 +21,12 @@ int test_omp_aligned_alloc_on_device() {
   int errors = 0;
   
   omp_memspace_handle_t  memspace = omp_default_mem_space;
-  omp_alloctrait_t       traits[1] = {{omp_atk_alignment, 64}};
-  omp_allocator_handle_t alloc = omp_init_allocator(memspace,1,traits);
+  const omp_alloctrait_t traits[1] = {{omp_atk_alignment, 64}};
+  omp_alloctrait_t       traits2[1] = {{omp_atk_alignment, 8}};
+
+  /* Host-side allocator; not used in the target region, which has a private
+     copy, initialized by uses_allocators.  */
+  omp_allocator_handle_t alloc = omp_init_allocator(memspace,1,traits2);
 
   #pragma omp target map(tofrom: errors) uses_allocators(alloc(traits)) 
   {

